build-bot-status.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+import sys
+import time
+import threading
+
+import requests
+
+from light_controller import LightController
+
+
+class BuildBotSession:
+    failColour = "red"
+    busyColour = "orange"
+    goodColour = "green"
+    exceptionColour = "purple"
+
+    def __init__(self, builderURL, slaveName):
+        self.builderURL = builderURL
+        self.slaveName = slaveName
+        self._statusURL = "{}/json/builders/{}".format(self.builderURL,
+                                                       self.slaveName)
+
+        self.lightController = LightController("/dev/ttyACM0",
+                                               "data/buttons.json")
+        self.lightController.lightsOn()
+        self.lightController.setColour('white')
+
+    def __del__(self):
+        self.lightController.lightsOff()
+
+    def start(self):
+        """
+        Poll as a thread
+        """
+        poller = threading.Thread(
+            target=self.getStatus)
+        poller.run()
+        poller.join()
+
+    def getStatus(self):
+        """
+        Get the general status of the builder
+        Runs continuously, polls every 60 seconds
+        """
+        while True:
+            status = requests.get(self._statusURL)
+            returned = status.json()
+            state = returned['state']
+            if state == 'idle':
+                self.getBuildStatus()
```

Comment:
Should we change the name to something like setIdleBuildStatus, instead of getBuildStatus, is now it is not clear that the method is actually setting the status of the light